### DISCUSSION TITLE
singleConnection PUT followed by GET

### DISCUSSION THREAD

**Discussion by dastein1** [enisdenjo/graphql-sse#81] [Q&A] [answered]
I'm trying to wrap some SSE api to be compatible with graphql-sse. With `singleConnection: false` this is trivial.
Now I'd like to get `singleConnection: true` to work. As per the doc I implemented the PUT to respond with some token.

    if (request.method === 'PUT') {
      return reply.status(201).headers({
        'content-type': 'text/plain; charset=utf-8',
      }).send(uid());
    }

But what I'm now seeing is that the client issues a PUT followed by a GET and then a POST request.
The GET request has only the header `x-graphql-event-stream-token` populated. What am I supposed to return here?

**Comment by enisdenjo** [comment, marked as answer]
The PUT request is to acquire the reservation token, then the GET is the actual event stream connection, and finally the POST is  executing operations (it contains the request parameters and the reservation token in the `extensions` field). All operation results are then streamed over the single GET event stream connection. [See how the event stream messages should be formatted.](https://github.com/enisdenjo/graphql-sse/blob/master/PROTOCOL.md#event-stream-1)

There's also a third request DELETE which is for [stopping active subscriptions](https://github.com/enisdenjo/graphql-sse/blob/master/PROTOCOL.md#stopping-streaming-operations).

**Reply by dastein1** [reply]
Ah, damn I got this completely wrong. So on the POST request I can simply grap the `operationId` and return some 202 response. When I then want to yield a message onto the stream I just set the `id` to that `operationId`.

**Reply by enisdenjo** [reply]
>  So on the POST request I can simply grap the operationId and return some 202 response. When I then want to yield a message onto the stream I just set the id to that operationId.

Exactly! You're getting the hang of it. 😄
